java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1952,6 +1946,30 @@ public final Builder append(double value) {
       return this;
     }
 
+    public final Builder append(String value) {
+      assert value != null;
+      return appendUTF8String(value.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private Builder appendUTF8String(byte[] value) {
+      return appendUTF8String(value, 0, value.length);
+    }
+
+    private Builder appendUTF8String(byte[] value, int offset, int length) {
+      assert value != null;
+      assert offset >= 0;
+      assert length >= 0;
+      assert value.length + offset <= length;
+      assert type == DType.STRING_CATEGORY || type == DType.STRING;
+      assert currentIndex < rows;
+      assert (currentStringByteIndex + length) <= stringBufferSize;
+      data.setBytes(currentStringByteIndex, value, offset, length);
+      currentStringByteIndex += length;
+      offsets.setInt((currentIndex + 1) * 4, currentStringByteIndex);
```

Comment:
Nit: Can move the currentIndex increment to just before this line to simplify this.